lib/rucio/client/ruleclient.py

```diff
@@ -192,7 +218,9 @@ def deny_replication_rule(self, rule_id):
         exc_cls, exc_msg = self._get_exception(headers=r.headers, status_code=r.status_code, data=r.content)
         raise exc_cls(exc_msg)
 
-    def list_replication_rule_full_history(self, scope, name):
+    def list_replication_rule_full_history(
+            self, scope: Union[str, bytes], name: Union[str, bytes]
```

Comment:
Why `bytes`?

Reply:
scope and name are used as arguments for the `quote_plus` function, which takes `str` or `bytes`. This was the only upper bound to the types applicable to scope and name I could find.